app/helpers/application_helper.rb

```diff
@@ -22,4 +22,11 @@ def render_flash_message(flash_value)
     end
   end
 
+  def user_has_membership_application?
+    current_user.try(:membership_applications).try(:any?) && current_user.try(:membership_applications).try(:last).try(:persisted?)
+  end
+
+  def member_has_company?
+    current_user.is_member && !current_user.membership_applications.last.company.nil?
+  end
```

Comment:
A user is only a member (`is_member == true`) if they have submitted an application and the application has been approved.

I think the really issue is that we haven't actually implemented the logic to do that processing.  (Hence the feature has to set things up such that all of those relations and models are valid.)